converter.go

```diff
@@ -762,3 +763,306 @@ func higherPrecisionEnabled(ctx context.Context) bool {
 	d, ok := v.(bool)
 	return ok && d
 }
+
+func arrowToRecord(record array.Record, rowType []execResponseRowType) (array.Record, error) {
+	s, err := recordToSchema(record.Schema(), rowType)
+	if err != nil {
+		return nil, err
+	}
+
+	var cols []array.Interface
+	numRows := record.NumRows()
+	pool := memory.NewGoAllocator()
+
+	for i, col := range record.Columns() {
+		srcColumnMeta := rowType[i]
+		data := col.Data()
+		converted := false
```

Comment:
similarly to the schema comment, could you make a `newCol` that points to `col` and get overridden when building a new one + one append at the end of the case?